neuro_san/service/usage/wrapped_usage_logger.py

```diff
@@ -0,0 +1,130 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import List
+
+from asyncio import run
+from os import environ
+
+from neuro_san.interfaces.usage_logger import UsageLogger
+
+
+class WrappedUsageLogger(UsageLogger):
+    """
+    Implementation of the UsageLogger interface that wraps another UsageLogger
+    and whose log_usage method makes sure sloppier inputs adhere better to the
+    UsageLogger contract.
+    """
```

Comment:
Class for easier server access to any given UsageLogger implementation.
Also handles the case gracefully when there is no UsageLogger desired.